poseidon2-air/src/air.rs

```diff
@@ -243,81 +263,32 @@ fn eval_sbox<AB, const DEGREE: usize, const REGISTERS: usize>(
 ) where
     AB: AirBuilder,
 {
-    // assert_ne!(REGISTERS, 0, "The number of REGISTERS must be positive.");
-    // assert!(DEGREE <= 11, "The DEGREE must be less than or equal to 11.");
-    // assert_eq!(
-    //     REGISTERS,
-    //     Self::OPTIMAL_REGISTER_COUNT[DEGREE],
-    //     "The number of REGISTERS must be optimal for the given DEGREE."
-    // );
-
-    let x2 = x.square();
-    let x3 = x2.clone() * x.clone();
-    load(sbox, 0, x3.clone(), builder);
-    if REGISTERS == 1 {
-        *x = sbox.0[0].into();
-        return;
-    }
-    if DEGREE == 11 {
-        (1..REGISTERS - 1).for_each(|j| load_product(sbox, j, &[0, 0, j - 1], builder));
-    } else {
-        (1..REGISTERS - 1).for_each(|j| load_product(sbox, j, &[0, j - 1], builder));
+    *x = match (DEGREE, REGISTERS) {
+        (3, 0) => x.cube(),
+        (5, 0) => x.exp_const_u64::<5>(),
+        (7, 0) => x.exp_const_u64::<7>(),
```

Comment:
I think it could be good to support them as options for anyone using higher-degree constraint systems, like although most Plonky3 users are using degree, most Plonky2 users were using degree 9